### PR TITLE
Retainbug

### PR DESCRIPTION
**Pull request by ptersilie**
This PR fixes two bugs found by the fuzzy tester. Might need rebasing, as I'm unsure if I should keep the commits separate or not.

### CODE REVIEW

File: lib/eco/incparser/astree.py [hunk 1]
```diff
@@ -222,10 +222,15 @@ def update_siblings(self):
         for i in range(len(self.parent.children)):
             c = self.parent.children[i]
             if c is self:
+
```
Comment (ltratt) [inline lib/eco/incparser/astree.py]: Random blank line?

**Comment by ltratt** [comment]
One minor comment to address. The two commits should be separate IMHO (after the comment commit has been squashed away).

File: lib/eco/incparser/astree.py (hunk 1 above)
Comment (ptersilie) [reply lib/eco/incparser/astree.py]: fixed

**Comment by ltratt** [comment]
OK, if you squash that last commit into whichever of the other two it should be part of, then I think I can merge.

**Comment by ptersilie** [comment]
Squashed!

**Review by ltratt — APPROVED** [review]

**Pull request merged**